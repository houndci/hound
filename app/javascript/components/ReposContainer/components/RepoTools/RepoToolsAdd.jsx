import React from 'react';

const RepoToolsAdd = (props) => (
  <div className="repo-tools-add">
    <a
      className="button repo-tools-add-button"
      href={`https://github.com/apps/${props.appName}/installations/new`}
    >
      Add GitHub repos
    </a>
  </div>
);

export default RepoToolsAdd;
