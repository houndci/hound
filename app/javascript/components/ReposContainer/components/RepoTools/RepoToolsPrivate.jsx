import React from 'react';

export default class RepoToolsPrivate extends React.Component {
  render() {
    const { onPrivateClicked } = this.props;

    return (
      <div className="repo-tools-private">
        <form method="post" action="/auth/github?access=full">
          <button
            className="repo-tools-private-button"
            type="submit"
            onClick={onPrivateClicked}
          >
            <span>Include private repos</span>
          </button>
        </form>
      </div>
    );
  }
}
