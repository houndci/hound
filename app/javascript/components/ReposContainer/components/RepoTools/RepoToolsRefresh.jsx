import React from 'react';

export default class RepoToolsRefresh extends React.Component {
  buttonText(isSyncing) {
    if (isSyncing) {
      return Hound.settings.syncingButtonText;
    } else {
      return Hound.settings.syncNowButtonText;
    }
  }

  render() {
    const { isSyncing, onRefreshClicked } = this.props;

    return (
      <div className="repo-tools-refresh">
        <button
          className="repo-tools-refresh-button"
          disabled={isSyncing ? "disabled" : null}
          onClick={onRefreshClicked}
        >
          <span>{this.buttonText(isSyncing)}</span>
        </button>
      </div>
    );
  }
}
