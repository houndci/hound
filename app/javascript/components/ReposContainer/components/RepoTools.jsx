import React from 'react';

import RepoToolsSearch from './RepoTools/RepoToolsSearch';
import RepoToolsRefresh from './RepoTools/RepoToolsRefresh';
import RepoToolsPrivate from './RepoTools/RepoToolsPrivate';

export default class RepoTools extends React.Component {
  render() {
    const {
      onSearchInput,
      showPrivateButton,
      isSyncing,
      onRefreshClicked,
    } = this.props;

    return (
      <div className="repo-tools">
        <RepoToolsSearch onSearchInput={onSearchInput} />
        {showPrivateButton ? <RepoToolsPrivate /> : null}
        <RepoToolsRefresh
          isSyncing={isSyncing}
          onRefreshClicked={onRefreshClicked}
        />
      </div>
    );
  }
}
