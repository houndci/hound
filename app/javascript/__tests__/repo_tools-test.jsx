import RepoTools from '../components/ReposContainer/components/RepoTools';

it('renders appropriately without Show Private button (not syncing)', () => {
  const has_private_access = true;

  const wrapper = shallow(
    <RepoTools
      showPrivateButton={!has_private_access}
      onSearchInput={(event) => jest.fn()}
      onRefreshClicked={(event) => jest.fn()}
      onPrivateClicked={(event) => jest.fn()}
      isSyncing={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately with Show Private button (not syncing)', () => {
  const has_private_access = false;

  const wrapper = shallow(
    <RepoTools
      showPrivateButton={!has_private_access}
      onSearchInput={(event) => jest.fn()}
      onRefreshClicked={(event) => jest.fn()}
      onPrivateClicked={(event) => jest.fn()}
      isSyncing={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
