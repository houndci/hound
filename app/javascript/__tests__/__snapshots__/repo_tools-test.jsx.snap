// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders appropriately with Show Private button (not syncing) 1`] = `
<div
  className="repo-tools"
>
  <RepoToolsSearch
    onSearchInput={[Function]}
  />
  <RepoToolsPrivate />
  <RepoToolsRefresh
    isSyncing={false}
    onRefreshClicked={[Function]}
  />
</div>
`;

exports[`renders appropriately without Show Private button (not syncing) 1`] = `
<div
  className="repo-tools"
>
  <RepoToolsSearch
    onSearchInput={[Function]}
  />
  <RepoToolsRefresh
    isSyncing={false}
    onRefreshClicked={[Function]}
  />
</div>
`;
